app/uk/gov/hmrc/uknwauthcheckerapi/services/ValidationService.scala

```diff
@@ -36,28 +39,38 @@ class ValidationService {
       case errors: JsError => Left(ValidationDataRetrievalError(errors))
     }
 
-  private def validateAuthorisationRequest(request: AuthorisationRequest): Either[JsError, AuthorisationRequest] = {
-    val eoris = request.eoris
-
-    val eoriErrors: Seq[JsonValidationError] = eoris
-      .filterNot(e => e matches CustomRegexes.eoriPattern)
-      .map(e => JsonValidationError(s"$e is not a supported EORI number"))
-
-    (eoriErrors.nonEmpty, isEoriSizeInvalid(eoris.size)) match {
-      case (false, false) => Right(request)
-      case (_, true)      => Left(JsError(JsPath \ "eoris", ErrorMessages.invalidEoriCount))
-      case (true, _) =>
-        Left(
-          JsError(
-            Seq("eoris").map { field =>
-              (JsPath \ field, eoriErrors)
-            }
-          )
+  private def validateAuthorisationRequest(request: AuthorisationRequest): Either[JsError, AuthorisationRequest] =
+    for {
+      _ <- validateEoriCount(request)
+      _ <- validateEoriStructure(request)
+    } yield request
+
+  private def validateEoriCount(request: AuthorisationRequest): ValidationResult[AuthorisationRequest] =
+    if (isEoriSizeInvalid(request.eoris.size)) {
+      Left(JsError(JsPath \ JsonPaths.eoris, ApiErrorMessages.invalidEoriCount))
+    } else {
+      Right(request)
+    }
+
+  private def validateEoriStructure(request: AuthorisationRequest): ValidationResult[AuthorisationRequest] = {
+    val eoriErrors = request.eoris.collect {
+      case eori if !(eori matches CustomRegexes.eoriPattern) => JsonValidationError(s"$eori is not a supported EORI number")
```

Comment:
What do you think about also adding `$eori is not a supported EORI number` to ApiErrorMessages as well to follow the other ones?

To keep it as vals you could do something like
` val invalidEori: String => String = eori => s"$eori is not a supported EORI number"`, or have it as a def either or
then just have 
`case eori if !(eori matches CustomRegexes.eoriPattern) => JsonValidationError(ApiErrorMessages.invalidEori(eori)`

Reply:
Good shout, done.